idmapping-rest/src/main/java/org/uniprot/api/idmapping/repository/IdMappingRepository.java

```diff
@@ -0,0 +1,88 @@
+package org.uniprot.api.idmapping.repository;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.Size;
+
+import lombok.AllArgsConstructor;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.params.MapSolrParams;
+import org.springframework.stereotype.Repository;
+import org.uniprot.api.idmapping.model.IdMappingStringPair;
+import org.uniprot.store.search.SolrCollection;
+
+@Repository
+@AllArgsConstructor
+public class IdMappingRepository {
+
+    private static final int MAX_ID_MAPPINGS_ALLOWED = 100_000;
+    private final SolrClient uniProtKBSolrClient;
```

Comment:
using @SiZe was just a hint for developer. But if it is casuing problem i will remove it
adding MAX_ID_MAPPINGS_ALLOWED in code will make hard code dependency so that if some one change id.mapping.max.from.ids.count property, will not break code.
there is limit of request size which we can send to solr.
there might be better way to put that intent in code, we can dissuss on it